resolvers.js

```diff
@@ -157,19 +157,31 @@ const Resolvers = {
     },
 
     opticalImageUrl(_, {datasetId, zoom}) {
-      const intZoom = zoom <= 1.5 ? 1 : (zoom <= 3 ? 2 : (zoom <= 6 ? 4 : 8));
-      return pg.select().from('optical_image')
-               .where('ds_id', '=', datasetId)
-               .where('zoom', '=', intZoom)
-               .then(records => {
-                 if (records.length > 0)
-                   return '/optical_images/' + records[0].id;
-                 else
-                   return null;
-               })
-               .catch((e) => {
-                 logger.error(e);
-               })
+      // zoom = -1 a trick to get all optical images IDs for a single dataset
+      if (zoom == -1) {
```

Comment:
Instead of introducing an obscure magic number "-1" here, I'd change the `schema.graphql` to remove default value for `zoom` (it's `1` currently), thereby allowing it to be `undefined`, and add a comment to the schema that `undefined` is supposed to return image IDs for all zoom levels.

Reply:
Alternatively, one may simply move this new code into another method, since the return type is actually different in this case: array of strings vs string. At least I don't see any advantage of having these two queries hidden behind the same GraphQL call

Reply:
@intsco hmm, that's true, I just thought that there is a difference in what each methods does. Basically, 
- deleteOpticalImage
fetches POST query to sm-engine, in which afterwards del_optical_image method is called to delete all entries in DB 
- opticalImageUrl 
just returns strings of image IDs in optical_image folder (in case of zoom=-1) that are further handled by web-app, which sends for every ID DELETE request to be deleted from File system (handled by Express app.delete). I think in any case retrieving all ids is needed here, or maybe there is a better way for that?

@iprotsyuk another method not to mess up it with the current one is a good idea, thanks! I also though of removing value for zoom to undefined, but then I thought that it may cause some unpredictable behavior, that's why I decided to pick -1.

Reply:
@Ren22, I'd suggest concentrating all deletion actions at one place, which sm-engine (`del_optical_image` hanlder) seems to be a best place for. Otherwise, one single action of image removal gets too much distributed